tutorials/intermediate/blitting.py

```diff
@@ -0,0 +1,186 @@
+"""
+=================
+Blitting Tutorial
+=================
+
+'Blitting' is a `standard technique
+<https://en.wikipedia.org/wiki/Bit_blit>`__ in computer graphics that
+in the context of Matplotlib can be used to (drastically) improve
+performance of interactive figures.  It is used internally by the
+:mod:`~.animation` and :mod:`~.widgets` modules for this reason.
+
+The source of the performance gains is simply not re-doing work we do
+not have to.  For example, if the limits of an Axes have not changed,
+then there is no reason we should re-draw all of the ticks and
+tick-labels (particularly because text is one of the more expensive
+things to render).
+
+The procedure to save our work is roughly:
+
+- draw the figure, but exclude any artists marked as 'animated'
+- save a copy of the Agg RBGA buffer
+
+In the future, to update the 'animated' artists we
+
+- restore our copy of the RGBA buffer
+- redraw only the animated artists
+- show the resulting image on the screen
+
+thus saving us from having to re-draw everything which is _not_
+animated.
+
+Simple Example
+--------------
+
+We can implement this via methods on `.FigureCanvasAgg` and setting
+``animated=True`` on our artist.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+x = np.linspace(0, 2 * np.pi, 100)
+
+fig, ax = plt.subplots()
+# animated=True makes the artist be excluded from normal draw tree
+(ln,) = ax.plot(x, np.sin(x), animated=True)
+
+# stop to admire our empty window axes and ensure it is drawn
+plt.pause(0.1)
+
+# save a copy of the image sans animated artist
```

Comment:
People coming here from `Animation` will probably not know what `Bbox`s are, and I don't think it adds too much noise to implicitly mention that this trick works for arbitrary regions of the figure.

```suggestion
# get copy of entire figure (everything inside fig.bbox) sans animated artist
```